tracer-core/src/main/java/org/zalando/tracer/Random64Generator.java

```diff
@@ -12,11 +12,38 @@
 @API(status = EXPERIMENTAL)
 public final class Random64Generator implements Generator {
 
+    private static final char[] DIGITS = {
+            '0', '1', '2', '3', '4', '5', '6', '7',
+            '8', '9', 'a', 'b', 'c', 'd', 'e', 'f',
+    };
+
     @Override
     public String generate() {
-        final ThreadLocalRandom random = ThreadLocalRandom.current();
-        // set most significant bit to produce fixed length string
-        return Long.toHexString(random.nextLong() | Long.MIN_VALUE);
+        ThreadLocalRandom random = ThreadLocalRandom.current();
+
+        char[] chars = new char[16];
+
+        long2hex(chars, 0, random.nextLong());
+
+        return new String(chars);
     }
 
+    protected static void long2hex(char[] chars, int offset, long r) {
```

Comment:
I'd move this somewhere else. Package private `Hex` class, maybe? `void Hex.fill(char[], int, long)`?